PullRequestClosed/PullRequestClosed.test.ts

```diff
@@ -0,0 +1,150 @@
+      return {
+        data: [
+          { body: "Renovate Jira issue ID: KEY-1234" }
+        ]
+      };
+    };
+  }
+  await action.run();
+}
+
+describe('PullRequestClosed', () => {
+  const originalKeys = Object.keys(process.env);
+  let logTester: LogTester;
+
+  beforeEach(() => {
+    logTester = new LogTester();
+    for (const key of Object.keys(process.env)) {
+      if (!originalKeys.includes(key)) {
+        // Otherwise, changes form previous UT are propagated to the next one
+        delete process.env[key];
+      }
+    }
+    process.env['GITHUB_REPOSITORY'] = 'test-owner/test-repo';
+    process.env['INPUT_GITHUB-TOKEN'] = 'fake';
+    github.context.payload = {
+      pull_request: {
+        number: 42
+      }
+    };
+  });
+
+  afterEach(() => {
+    logTester.afterEach();
+  });
+
+  it('is-eng-xp-squad non-Bot PR skips issue resolution', async () => {
+    process.env['INPUT_IS-ENG-XP-SQUAD'] = 'true';
+    await runAction('KEY-1234 Title');
+    expect(logTester.logsParams).toStrictEqual([
+      "Loading PR #42",
+      "Loading PR #42",
+      "Skipping issue resolution for non-Bot PR",
+      "Done"
+    ]);
+  });
+
+  it('is-eng-xp-squad Bot PR merged PR moves issue to Done', async () => {
+    process.env['INPUT_IS-ENG-XP-SQUAD'] = 'true';
+    github.context.payload.pull_request.merged = true;
+    github.context.payload.pull_request.base = { ref: 'master' };
+    await runAction('Title', "renovate[bot]");
+    expect(logTester.logsParams).toStrictEqual([
+      "Loading PR #42",
+      "Loading PR #42",
+      "Invoked jira.moveIssue('KEY-1234', 'Resolve issue', {\"resolution\":{\"id\":\"10000\"}})",
+      "Done",
+    ]);
+  });
+
+  it('is-eng-xp-squad Bot PR unmerged PR cancels issue', async () => {
+    process.env['INPUT_IS-ENG-XP-SQUAD'] = 'true';
+    await runAction('Title', "renovate[bot]");
+    expect(logTester.logsParams).toStrictEqual([
+      "Loading PR #42",
+      "Loading PR #42",
+      "Invoked jira.moveIssue('KEY-1234', 'Resolve issue', {\"resolution\":{\"id\":\"10001\"}})",
+      "Done",
+    ]);
+  });
+
+  it('Unmerged PR for ticket created by automation closed', async () => {
+    await runAction('KEY-5678 Title');
+    expect(logTester.logsParams).toStrictEqual([
+      "Loading PR #42",
+      "Invoked jira.moveIssue('KEY-5678', 'Cancel Issue', {\"resolution\":{\"id\":\"10001\"}})",
+      "Done"
+    ]);
+  });
+
+  it('Unmerged PR for pre-existing issue closed', async () => {
+    await runAction('KEY-1234 Title');
+    expect(logTester.logsParams).toStrictEqual([
+      "Loading PR #42",
+      "Skipping issue cancellation for creator Creator of KEY-1234",
+      "Done"
+    ]);
+  });
+
+  it('Jira issue does not exist', async () => {
+    const logSpy = jest.spyOn(core, 'setFailed').mockImplementation(() => { });
+    try {
+      await runAction('FAKE-1234 Title');
+      expect(logSpy).toHaveBeenCalledWith('Scaffolding did not expect FAKE-1234');
```

Comment:
Nope, you should have updated the scaffolding to case `FAKE-1234` and return `null`. The current version doesn't test the action, it tests the scaffolding